File: Python/maxsumarr.py
#coded w/ <3 by inf1nity
input()
n = list(map(int, input().split(' ')))
k = int(input())
mx = 0
for i in range(len(n)-1):
    for j in range(i+1, len(n)):
        if sum(n[i:j])>mx and sum(n[i:j])<=k:
            mx = sum(n[i:j])
print(mx)

File: Pascal/maxsumarr.pas
//coded w/ <3 by inf1nity
var a:array[0..100000000] of int64;
    i,j,n,mx,k:longint;
//////////////////////////////////
begin
readln(n);
for i:=1 to n do
    begin
    read(a[i]);
    a[i]:=a[i]+a[i-1];
    end;
readln(k);
for i:=1 to n-1 do
    for j:=i+1 to n do
        if (a[j]-a[i-1]>mx) and (a[j]-a[i-1]<=k) then
            mx:=a[j]-a[i-1];
writeln(mx);
end.